src/example/SampleCustomLoader/sampleCustLoader.cpp

```diff
@@ -0,0 +1,419 @@
+        std::ifstream fileToRead(fileName);
+        std::string stateStr;
+        if (fileToRead.is_open()) {
+            getline(fileToRead, stateStr);
+        }
+
+        if (stateStr == "DISABLED") {
+            std::cout << "Balcklisting Model:: " << it.first.first << std::endl;
+            models_blacklist_local.insert({it.first, true});
+        }
+    }
+
+    // Now take the mutex and copy to original map
+    std::lock_guard<std::mutex> mapGuard(map_mutex);
+    models_blacklist.clear();
+    models_blacklist.insert(models_blacklist_local.begin(), models_blacklist_local.end());
+}
+
+void custSampleLoader::startWatcher(int interval) {
+    watchIntervalSec = interval;
+
+    if ((!watcherStarted) && (watchIntervalSec > 0)) {
+        std::thread th(std::thread(&custSampleLoader::threadFunction, this));
+        watcher_thread = std::move(th);
+        watcherStarted = true;
+    }
+    std::cout << "custSampleLoader: StartWatcher" << std::endl;
+}
+
+void custSampleLoader::watcherJoin() {
+    std::cout << "custSampleLoader: watcherJoin()" << std::endl;
+    if (watcherStarted) {
+        if (watcher_thread.joinable()) {
+            watcherStarted = false;
+            cv.notify_all();
+            watcher_thread.join();
+        }
+    }
+}
+
+/* 
+ * From the custom loader options extract the mdeol file name and other neede information and
+ * load the model and optional bin file into buffers and return
+ */
+CustomLoaderStatus custSampleLoader::loadModel(const std::string& modelName, const std::string& basePath, const int version,
+    const std::string& loaderOptions, std::vector<uint8_t>& model, std::vector<uint8_t>& weights) {
+    std::cout << "custSampleLoader: Custom loadModel" << std::endl;
+
+    std::string binFile;
+    std::string modelFile;
+    std::string enableFile;
+    int modelType = SAMPLE_LOADER_IR_MODEL;
+    CustomLoaderStatus st = CustomLoaderStatus::MODEL_LOAD_ERROR;
+
+    int ret =
+        extract_input_params(basePath, version, loaderOptions, binFile, modelFile, enableFile, modelType);
+    if (ret != SAMPLE_LOADER_OK) {
+        std::cout << "custSampleLoader: Invalid custom loader options" << std::endl;
+        return st;
+    }
+
+    // load models
+    ret = load_files(binFile, modelFile, modelType, model, weights);
+    if (ret != SAMPLE_LOADER_OK) {
+        std::cout << "custSampleLoader: Could not read model files" << std::endl;
+        return CustomLoaderStatus::INTERNAL_ERROR;
+    }
+
+    /* Start the watcher thread after first moel load */
+    if (watcherStarted == false) {
+        startWatcher(MODEL_CHECK_PERIOD);
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    }
+
+    auto modelId = std::make_pair(modelName, version);
+    models_loaded.emplace_back(modelId);
+
+    if (!(enableFile.empty())) {
+        auto it = models_watched.find(modelId);
+        if (it == models_watched.end()) {
+            models_watched[modelId] = enableFile;

```

Comment:
Minor:
you are creating empty std::string in map and then replace it with enableFile. You could use std::map.insert/emplace with std::move

Reply:
models_watched is used to check the status of the model if it is loaded or unloaded as long as it is in config file. Here, I am adding the entry to models_watched only if the "enable_file" is specified in the config file. After your comments, realized that, actually I need to change this part of code to handle the case where enable_file parameter is changed in config file and that is passed a fresh. In that case we need to check new file to check model status.